examples/systemcalls/systemcalls.c

```diff
@@ -94,6 +147,30 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
 */
 
     va_end(args);
+    
+    if (command[0][0] != '/') {
+        return false; // Ensure absolute path
+    }
+
+    pid_t pid = fork();
+    if (pid == -1) {
+        return false;
+    }
+    else if (pid == 0) {
+        int fd = open(outputfile, O_WRONLY | O_CREAT | O_TRUNC, 0644);
+        if (fd == -1) {
+            return false;
+        }
+        dup2(fd, STDOUT_FILENO);
+        close(fd);
+        execv(command[0], command);
+        return false;
```

Comment:
Rather than `return` here you should be calling `exit`.  As written, the child process will continue to run after failure, and the failure the parent returns is the failure of the overall child process.  Your check for absolute path above is likely obscuring this case by failing early and not requiring the child to properly handle execv failures.